### PR TITLE
Add pipelines-as-code for enterprise contract

### PR DESCRIPTION
**Pull request by mbharatk**
Tekton is adopted as a primary CI and CD ( images builds ) for enterprise-contract

### CODE REVIEW

File: components/enterprise-contract/.tekton/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pipeline
+secrets:
+  - name: redhat-appstudio-staginguser-pull-secret
```
Comment (Michkov) [inline components/enterprise-contract/.tekton/serviceaccount.yaml]: this file is not needed anymore, secrets are handled by sharedsecret.

File: components/enterprise-contract/kustomization.yaml [hunk 2]
```diff
@@ -3,3 +3,4 @@ kind: Kustomization
 
 resources:
 - https://github.com/hacbs-contract/enterprise-contract-controller/config/crd?ref=69e8c88a50c471acca37a7eff68e9d129dad08e6
+- .tekton/
```
Comment (Michkov) [inline components/enterprise-contract/kustomization.yaml:6]: currently no namespace is defined where the resources should be created.

File: components/enterprise-contract/.tekton/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+resources:
+- serviceaccount.yaml
```
Comment (Michkov) [inline components/enterprise-contract/.tekton/kustomization.yaml:2]: no such file

File: components/enterprise-contract/.tekton/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+- pipelines-as-code.yaml
+- pvc.yaml
+
```
Comment (Michkov) [inline components/enterprise-contract/.tekton/kustomization.yaml:5]: Update https://github.com/redhat-appstudio/infra-deployments/blob/main/components/build/shared-resources/shared-resources-components.yaml to allow namespace used by this service to access the secrets.

**Review by Michkov — CHANGES_REQUESTED** [review]

**Comment by lcarva** [comment]
@mbharatk, I don't quite follow what is being achieved by this PR. Can you explain it to me (in simple terms :smile: ) ?

**Comment by Michkov** [comment]
@lcarva context -> [PLNSRVCE-270](https://issues.redhat.com/browse/PLNSRVCE-270). We are applying Pipelines-as-code for each repostiory. But in this case it would need some discussion why hacbs-contract is using own repositories instead of redhat-appstudio ones.

**Review by Michkov — APPROVED** [review]

**Review by zregvart — APPROVED** [review]
LGTM

**Pull request merged**